examples/go/README.md

```diff
@@ -0,0 +1,92 @@
+# Go example
+
+This is an example of how to migrate from
+[`go_image`](https://github.com/bazelbuild/rules_docker#go_image) in rules_docker.
+
+## Base image
+
+First, you'll need a base image.
+
+It's wise to minimize changes by using the same one your current `go_image` uses.
+The logic for choosing the default base in rules_docker is in
+[https://github.com/bazelbuild/rules_docker/blob/fc729d85f284225cfc0b8c6d1d838f4b3e037749/go/image.bzl#L114](go_image.bzl)
+
+Or, you can just use bazel query to find out:
+
+```
+$ bazel query --output=build hello_world:hello_world_go_image
+Starting local Bazel server and connecting to it...
+# /home/alexeagle/Projects/hello_world/BUILD.bazel:22:9
+_app_layer(
+  name = "hello_world_go_image",
+  base = select({"@io_bazel_rules_docker//:debug": "@go_debug_image_base//image:image", "@io_bazel_rules_docker//:fastbuild": "@go_image_base//image:image", "@io_bazel_rules_docker//:optimized": "@go_image_base//image:image", "//conditions:default": "@go_image_base//image:image"}),
+...
+)
+```
+
+Since we don't use the "debug" config, this tells us that we use `@go_base_image`, we need one more lookup to see what that uses:
+
+```
+$ bazel query --output=build @go_image_base//image:image
+# /shared/cache/bazel/user_base/bf7b6accf6f1187bd5511f3fbf7b21b9/external/go_image_base/image/BUILD:4:17
+container_import(
+  name = "image",
+  base_image_registry = "gcr.io",
+  base_image_repository = "distroless/base",
+...
+)
+```
+
+Now that we know it's `gcr.io/distroless/base` we can pull the same base image by adding to WORKSPACE:
+
+```
+load("@contrib_rules_oci//oci:pull.bzl", "oci_pull")
+
+oci_pull(
+    name = "distroless_base",
+    digest = "sha256:ccaef5ee2f1850270d453fdf700a5392534f8d1a8ca2acda391fbb6a06b81c86",
+    image = "gcr.io/distroless/base",
+    platforms = ["linux/amd64","linux/arm64"],
```

Comment:
When I run the following command, I see different digests for the AMD64 and ARM64 images:
```shell
docker manifest inspect gcr.io/distroless/base \
  | jq --raw-output \
       '.manifests[] | select(.platform.os == "linux") | {arch: .platform.architecture, digest}'
```
What allows you to specify one digest here and assume that the resulting image will work for both of these machine architectures?

Reply:
crane has a digest command that lets you query the digest by running `crane digest <IMAGE>`.  alternatively, you could specify `tag` and `reproducible = False` attributes and replace the attributes according to the warning message oci_pull will print.

Reply:
This all makes more sense to me today. With Using a combination of _crane digest_, _docker manifest inspect_, and GCP's Web console to inspect the GCR entries, I now see how we're able to point at an image that contains a _manifest list_, with several entries, each for a different combination of operating system and machine architecture.

With the _docker manifest inspect_ command, it's not possible—or at least I couldn't figure out a way—to see the image digest for the image nominated by the tagged name. It dumps the manifest list, but doesn't tell you the digest of the image containing that manifest list. Using the _crane digest_ command closes that gap.

Thank you for the suggestion, @thesayyn.